packages/tinacms/src/admin/components/GetCollection.tsx

```diff
@@ -66,10 +67,85 @@ export const useGetCollection = (
       }
     }
 
+    const searchCollection = async () => {
```

Comment:
There's kind of a lot going on inside this useEffect. Might make sense to pull out some of these functions so we don't have to keep declaring them.

Reply:
I split the hooks up which hopefully makes things clearer